cpp/examples/string_transforms/output.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "common.hpp"
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/transform.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+std::unique_ptr<cudf::column> transform(cudf::table_view const& table)
+{
+  auto stream = rmm::cuda_stream_default;
+  auto mr     = cudf::get_current_device_resource_ref();
+
+  auto udf = R"***(
+  __device__ void email_provider(cudf::string_view* out,
+                                 cudf::string_view const email,
+                                 cudf::string_view const alt)
+  {
+    auto pos = email.find('@');
+  
+    if (pos == cudf::string_view::npos) {
+      *out = alt;
+      return;
+    }
+  
+    auto provider_begin = pos + 1;
+    auto provider       = email.substr(provider_begin, email.size_bytes() - provider_begin);
```

Comment:
```suggestion
    auto provider       = email.substr(provider_begin, email.length() - provider_begin);
```
`.length()` is the number of characters and `.size_bytes()` is the number of bytes.
The `.substr()` function expects values in units of characters.
The `provider_begin` is correct since `.find()` returns character position.
For your example `.length()==.size_bytes()` since email-providers are all ASCII characters but I'd rather this more be technically correct in case the code is copied for some other purpose.

Reply:
I'll change it to that, I keep forgetting it's a codepoint index